onmt/model_builder.py

```diff
@@ -102,6 +103,65 @@ def load_test_model(opt, model_path=None):
     return fields, model, model_opt
 
 
+def build_src_emb(model_opt, fields):
+    # Build embeddings.
+    if model_opt.model_type == "text":
+        src_field = fields["src"]
+        src_emb = build_embeddings(model_opt, src_field)
+    else:
+        src_emb = None
+    return src_emb
+
+
+def build_encoder_with_embeddings(model_opt, fields):
+    # Build encoder.
+    src_emb = build_src_emb(model_opt, fields)
+    encoder = build_encoder(model_opt, src_emb)
+    return encoder, src_emb
+
+
+def build_decoder_with_embeddings(
+    model_opt, fields, share_embeddings=False, src_emb=None
+):
+    # Build embeddings.
+    tgt_field = fields["tgt"]
+    tgt_emb = build_embeddings(model_opt, tgt_field, for_encoder=False)
+
+    if share_embeddings:
+        tgt_emb.word_lut.weight = src_emb.word_lut.weight
```

Comment:
This might break if `src_emb` is `None`.

Reply:
Do you have an example when this could happen ?
Can src_emb be None in seq2seq ? If it cannot, then it cannot be in LM either

Looking at the code it will happen when model_type is not "text" which is not supported yet

Reply:
Yes, no particular way this can happen right now. Just misleading with the kwarg defaulting to None in the definition.